fpdf/fpdf.py

```diff
@@ -1266,17 +1267,116 @@ def rect(self, x, y, w, h, style=None):
-        )
+
+        if round_corners is not False:
+            self._draw_rounded_rect(x, y, w, h, style, round_corners)
+        else:
+            style = RenderStyle.coerce(style)
+            self._out(
+                f"{x * self.k:.2f} {(self.h - y) * self.k:.2f} {w * self.k:.2f} "
+                f"{-h * self.k:.2f} re {style.operator}"
+            )
+
+    def _draw_rounded_rect(self, x, y, w, h, style, round_corners):
+        min = h
+        r = (w - h) / 2
+
+        if r < 0:
+            r *= -1
+            min = w
+
+        if r == 0:
+            r = w / 5
+
+        if r >= min / 2:
+            r /= min
+
+        point_1 = point_8 = (x, y)
+        point_2 = point_3 = (x + w, y)
+        point_4 = point_5 = (x + w, y + h)
+        point_6 = point_7 = (x, y + h)
+        coor_x = [x, x + w, x, x + w]
+        coor_y = [y, y, y + h, y + h]
+
+        if round_corners is True:
+            round_corners = [
+                Corner.TOP_RIGHT.value,
+                Corner.TOP_LEFT.value,
+                Corner.BOTTOM_RIGHT.value,
+                Corner.BOTTOM_LEFT.value,
+            ]
+
+        if Corner.TOP_RIGHT.value in round_corners:
+            self.arc(coor_x[0], coor_y[0], 2 * r, 180, 270, style=style)
+            point_1 = (x + r, y)
+            point_8 = (x, y + r)
+
+        if Corner.TOP_LEFT.value in round_corners:
+            self.arc(coor_x[1] - 2 * r, coor_y[1], 2 * r, 270, 0, style=style)
+            point_2 = (x + w - r, y)
+            point_3 = (x + w, y + r)
+
+        if Corner.BOTTOM_LEFT.value in round_corners:
+            self.arc(coor_x[3] - 2 * r, coor_y[3] - 2 * r, 2 * r, 0, 90, style=style)
+            point_4 = (x + w, y + h - r)
+            point_5 = (x + w - r, y + h)
+
+        if Corner.BOTTOM_RIGHT.value in round_corners:
+            self.arc(coor_x[2], coor_y[2] - 2 * r, 2 * r, 90, 180, style=style)
+            point_6 = (x + r, y + h)
+            point_7 = (x, y + h - r)
+
+        original_color = self.draw_color.colors
+        new_color = self.fill_color.colors
+
+        self.set_draw_color(new_color[0] * 255)
+        if style != "D":
+            if len(new_color) > 1:
+                self.set_draw_color(
+                    new_color[0] * 255, new_color[1] * 255, new_color[2] * 255
+                )
+
+            self.polyline(
+                [
+                    point_1,
+                    point_2,
+                    point_3,
+                    point_4,
+                    point_5,
+                    point_6,
+                    point_7,
+                    point_8,
+                    point_1,
+                ],
+                style=style,
+            )
+
+            self.set_draw_color(original_color[0])
+            if len(original_color) > 1:
+                self.set_draw_color(
+                    original_color[0], original_color[1], original_color[2]
+                )
+
+        if style != "F":
```

Comment:
This means that style is either `DRAW` or `DRAW_FILL`.
Once `style` is converted to a `RenderStyle`, you will be able to just call `.is_draw`.

Reply:
Fixed